matplotcheck/raster.py

```diff
@@ -74,22 +75,18 @@ def _which_label(self, label, all_label_options):
         string that is the first entry in the internal list which label is
         matched with. If no match is found, return value is None
         """
-        for label_opts in all_label_options:
-            for s in label_opts:
-                if s in label:
-                    return label_opts[0]
+        for label_option in all_label_options:
+            if label_option == label:
+                return label_option
         return None
 
-    def assert_legend_accuracy_classified_image(
-        self, im_expected, all_label_options
-    ):
+    def assert_raster_legend_labels(self, im_expected, all_label_options):
         """Asserts legend correctly describes classified image on Axes ax,
```

Comment:
Tests whether the legend labels on the input plot match labels that are expected.